app/lib/preservation_catalog/s3/audit.rb

```diff
@@ -4,52 +4,71 @@ module S3
-      end
+      delegate :bucket, :bucket_name, to: ::PreservationCatalog::S3
+
+      attr_reader :zmv, :results
 
-      # TODO: should we be capturing/reporting via AuditResults instance instead of just logging?  would be
-      # consistent with other audit checks.
-      def self.logger
-        @logger ||= Logger.new(Rails.root.join('log', 'c2a.log'))
+      # @param [ZippedMoabVersion] the ZippedMoabVersion to check
+      # @param [AuditResults] the AuditResults instance used to track findings for this audit run
+      def initialize(zmv, results)
+        @zmv = zmv
+        @results = results
       end
 
-      def self.check_aws_replicated_zipped_moab_version(zmv)
-        return unless check_child_zip_part_attributes(zmv)
+      # convenience method for instantiating the audit class and running the check in one call
+      def self.check_aws_replicated_zipped_moab_version(zmv, results)
+        new(zmv, results).check_aws_replicated_zipped_moab_version
+      end
 
+      def check_aws_replicated_zipped_moab_version
         zmv.zip_parts.where.not(status: :unreplicated).each do |part|
           aws_s3_object = bucket.object(part.s3_key)
-          if aws_s3_object.exists?
-            part.update(last_existence_check: Time.zone.now)
-          else
-            logger.error("Archival Preserved Copy: #{zmv.inspect} #{part.inspect} was not found on #{bucket_name}.")
-            part.update(status: 'not_found', last_existence_check: Time.zone.now)
-            next
-          end
+          next unless check_existence(aws_s3_object, part)
+          next unless compare_checksum_metadata(aws_s3_object, part)
+          part.ok!
+        end
+      end
 
-          # NOTE: no checksum computation is happening here (neither on our side, nor on AWS's).  we're just comparing
-          # the checksum we have stored with the checksum we asked AWS to store.  we really don't expect any drift, but
-          # we're here, and it's a cheap check to do, and it'd be weird if they differed, so why not?
-          # TODO: in a later work cycle, we'd like to spot check some cloud archives: that is, pull the zip down,
-          # re-compute the checksum for the retrieved zip, and make sure it matches what we stored.
-          replicated_checksum = replicated_checksum(aws_s3_object)
-          if part.md5 == replicated_checksum
-            part.update(last_checksum_validation: Time.zone.now)
-          else
-            logger.error("Stored checksum(#{part.md5}) doesn't match the replicated checksum(#{replicated_checksum}).")
-            part.update(status: 'replicated_checksum_mismatch', last_checksum_validation: Time.zone.now)
-            next
-          end
+      private
 
-          part.ok!
+      def compare_checksum_metadata(aws_s3_object, part)
+        # NOTE: no checksum computation is happening here (neither on our side, nor on AWS's).  we're just comparing
```

Comment:
You can move this comment block above the `def` to score some code-coverage points.